packages/opentelemetry-plugin-http/src/utils.ts

```diff
@@ -0,0 +1,214 @@
+      (options as RequestOptions).headers!.Expect
+    );
+  }
+
+  /**
+   * Check whether the given obj match pattern
+   * @param constant e.g URL of request
+   * @param obj obj to inspect
+   * @param pattern Match pattern
+   */
+  static satisfiesPattern<T>(
+    constant: string,
+    obj: T,
+    pattern: IgnoreMatcher<T>
+  ): boolean {
+    if (typeof pattern === 'string') {
+      return pattern === constant;
+    } else if (pattern instanceof RegExp) {
+      return pattern.test(constant);
+    } else if (typeof pattern === 'function') {
+      return pattern(constant, obj);
+    } else {
+      throw new TypeError('Pattern is in unsupported datatype');
+    }
+  }
+
+  /**
+   * Check whether the given request is ignored by configuration
+   * @param constant e.g URL of request
+   * @param obj obj to inspect
+   * @param list List of ignore patterns
+   */
+  static isIgnored<T>(
+    constant: string,
+    obj: T,
+    list?: Array<IgnoreMatcher<T>>
+  ): boolean {
+    if (!list) {
+      // No ignored urls - trace everything
+      return false;
+    }
+
+    for (const pattern of list) {
+      if (Utils.satisfiesPattern(constant, obj, pattern)) {
+        return true;
+      }
+    }
+
+    return false;
+  }
+
+  /**
+   * Will subscribe obj on error event and will set attributes when emitting event
+   * @param span to set
+   * @param obj to subscribe on error
+   */
+  static setSpanOnError(span: Span, obj: IncomingMessage | ClientRequest) {
+    obj.on('error', error => {
+      span.setAttributes({
+        [AttributeNames.ERROR]: true,
+        [AttributeNames.HTTP_ERROR_NAME]: error.name,
+        [AttributeNames.HTTP_ERROR_MESSAGE]: error.message,
+      });
+
+      let status: Status;
+      if ((obj as IncomingMessage).statusCode) {
+        status = Utils.parseResponseStatus(
+          (obj as IncomingMessage).statusCode!
+        );
+      } else {
+        status = { code: CanonicalCode.UNKNOWN };
+      }
+
+      status.message = error.message;
+
+      span.setStatus(status);
+      span.end();
+    });
+  }
+
+  /**
+   * Makes sure options is an url object
+   * return an object with default value and parsed options
+   * @param options for the request
+   */
+  static getRequestInfo(options: RequestOptions | string) {
+    let pathname = '/';
+    let origin = '';
+    let optionsParsed: url.URL | url.UrlWithStringQuery | RequestOptions;
+    if (typeof options === 'string') {
+      optionsParsed = url.parse(options);
+      pathname = (optionsParsed as url.UrlWithStringQuery).pathname || '/';
+      origin = `${optionsParsed.protocol || 'http:'}//${optionsParsed.host}`;
+    } else {
+      optionsParsed = options;
+      try {
+        pathname = (options as url.URL).pathname;
+        if (!pathname && options.path) {
+          pathname = url.parse(options.path).pathname || '/';
+        }
+        origin = `${options.protocol || 'http:'}//${options.host}`;
```

Comment:
besides `host` also `hostname` is supported (see https://nodejs.org/dist/latest-v12.x/docs/api/http.html#http_http_request_url_options_callback)